cluster/images/etcd/migrate/migrate.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+)
+
+const (
+	versionFilename        = "version.txt"
+	defaultPort     uint64 = 18629
```

Comment:
Hmm - I still don't see solution for port conflicts between main etcd and event etcd.

I think you should at least set "--port" flag when you're calling "migrate binary in the script.

Reply:
No need. Since etcd and etcd events are run in separate docker containers, and neither of them expose 18629 to the host operating system, they can concurrently use this port within the docker containers.

Reply:
@wojtek-t Ohh. Thanks so much for catching this!  I've populated the `--port` flag in `migrate-if-needed.sh`.